gk/main.c

```diff
@@ -654,6 +759,13 @@ cleanup_gk(struct gk_config *gk_conf)
 	int i;
 
 	for (i = 0; i < gk_conf->num_lcores; i++) {
+		unsigned socket = rte_lcore_to_socket_id(gk_conf->lcores[i]);
+		if (ipv6_if_configured(&gk_conf->net->front) &&
+				gk_conf->ipv6_acls[socket] != NULL) {
+			rte_acl_free(gk_conf->ipv6_acls[socket]);
+			gk_conf->ipv6_acls[socket] = NULL;
```

Comment:
Yes, we could have IPv6 ACLs for each interface instead of letting each block have their own. This would require a small adjustment though -- I'll describe it so that we are on the same page.

Right now, ACLs are allocated on a per-socket basis -- for example, each GK lcore doesn't need its own ACL, so all the lcores on the same socket share an ACL.

A problem arises when another block is trying to use the IPv6 ACL on the same socket that the GK lcores are using. The rules that this new block wants to add to the ACL may conflict with the rules installed by GK.

DPDK has a solution for this, which is that when classifying packets, it can return matches in different categories. So even if multiple blocks are using the same ACL for different rules, the classification algorithm can return one match for the GK-related rules and a match for the new block's rules.

I'll take this suggestion and the ones above to refactor the ACL stuff away from the blocks.